Add auth with certificates (#609)
hashicorp/web-unified-docs: content/well-architected-framework/docs/docs/secure-systems/secure-applications/authenticate-certificates/authenticate-workloads-tls.mdx

<ImageConfig hideBorder caption="Authenticate services with Consul and TLS certificates">

![Authenticate services with Consul and TLS certificates](/img/well-architected/diagram-secure-cluster-tls-consul.png#light-theme-only)

</ImageConfig>

<ImageConfig hideBorder>

![Authenticate services with Consul and TLS certificates](/img/well-architected/diagram-secure-cluster-tls-consul-dark.png#dark-theme-only)

</ImageConfig>

When you authenticate application workloads with Consul, you eliminate hard coded credentials and addresses. You also improve compliance through audit trails, monitoring, and centralized policy management. You can also define your policies as code, version control them, and quickly update policies as requirements change.

You can authenticate your application workloads on Consul with the following methods:

> - **ACL tokens:** Consul uses ACL tokens as its primary method of authentication. ACL tokens can take the form of static tokens or dynamic tokens backed by an auth method.

Reviewer: from my understanding, acl tokens is used more to authenticate to the consul cluster rather than the services. this is how you connect to consul to send it commands

```suggestion
```

> - **TLS certificates:** Consul supports TLS certificates to enhance security of application workloads with  service to service authentication.

Reviewer: ```suggestion
- **TLS certificates:** Consul uses mTLS certificates for service mesh to identify application services.
```

- **JWT (with Vault):** Consul can integrate with Vault to leverage JWT authentication for dynamic workload identities.

Keep in mind the following best practices when you authenticate your application workload with Consul:

> - **Principle of least privilege:** assign the minimal required capabilities and permissions to ACL tokens and TLS certificates.

Reviewer: ```suggestion
- **Principle of least privilege:** assign the minimal required capabilities and permissions to TLS certificates.
```

you dont auth application workloads with acl tokens in consul

> - **Regular rotation**: implement processes to regularly rotate ACL tokens.

Reviewer: ```suggestion
```
